Add APM experience Reference (#1071)
timescale/docs: promscale/visualize-data/apm-experience.md

## Importaing APM dashboards from Grafana community

All the dashboards are published [here](promscale-grafana-dashboards)

<procedure>

1.  Open the dashboard published by Promscale org from the above reference URL. Select 
    `Copy ID to Clipboard` to copy the dashboard UID.
1.  In Grafana UI select `Create` from left side menu bar, navigate to `Import`.
1.  Now paste the dashbaord UID in `Import via grafana.com` textbox and click 
    on `Load` to import the dashbaord into Grafana instance.
1.  On importing the dashboard select the `folder` to which you want to add the imported
    dashboard.
1.  Select the data source from which you want the dashboard to query the data.
    * For APM dashboards select the `TimescaleDB / PostgreSQL` data source 
      as `Promscale-SQL`.
    * For APM dashboards select the `Promscale Jaeger Tracing` data source 
      as `Promscale-Tracing`.
    * For Promscale metrics select the `Prometheus` data source as `Promscale-PromQL`.

</procedure>

## Importing dashbaords as JSON files.

All the Promscale dashboards are available at [Promscale dashboards](promscle-github-dashboards). 

<procedure>

> 1.  Copy the content from the `.json` file from the above referenced Promscale dashboards. 

Author: agreed.

# Application Performance Management (APM) with traces

> 

Reviewer: ```suggestion
```

Promscale provides an APM-like experience with traces data using SQL. Import the Grafana dashboards that are published by the Promscale team to get the APM with traces.

> ## Before you begin importing APM dashboards

Reviewer: ```suggestion
Before you begin importing APM dashboards:

```

* Add these data sources in Grafana:
    * [Jaeger data source][promscale-as-jaeger].
    * [PostgreSQL data source][promscale-as-postgresql].   

> * Ensure that timescaledb_toolkit PostgreSQL extension is installed.

Reviewer: ```suggestion
* Check that the `timescaledb_toolkit` extension is installed.
```

  To verify if the extension is installed, run this SQL query:
  `SELECT * FROM pg_extension WHERE extname='timescaledb_toolkit';` 
  If the query returns no results the extension is not installed.

> ## Installing timescaledb_toolkit PostgreSQL extension
> <procedure> 

Reviewer: ```suggestion
<procedure> 

## Installing the TimescaleDB toolkit extension
```

> To install the timescaledb_toolkit PostgreSQL extension:

Reviewer: ```suggestion

```

1. Check if the extension is available:

>    `SELECT * FROM pg_available_extensions WHERE name='timescaledb_toolkit';`

Reviewer: ```suggestion
1.  Check if the extension is available:

    ```sql
    SELECT * 
      FROM pg_available_extensions 
      WHERE name='timescaledb_toolkit';
    ```

```

>    <highlight type="note"> 
>    If the query returns no results, the extension is not available for installation
>    in your database. To make it available follow [these
>    instructions](install-toolkit). 
>    </highlight>

Reviewer: ```suggestion
     If the query returns no results, the extension is not available for installation
     in your database. To make it available follow 
     [these instructions](install-toolkit). 
```

> 1. Install the extension using: `CREATE EXTENSION timescaledb_toolkit;`

Reviewer: ```suggestion
1. Install the extension:

     ```sql
     CREATE EXTENSION timescaledb_toolkit;
     ```

```

</procedure>

You can use one of these methods to import APM dashboards:

*  From the Grafana community the dashboards published by Promscale.
*  From Promscale Github repository as JSON files.

> *  When installing Grafana, mount APM dashboards as a snapshot.

Reviewer: This might need to go further up the page, before the procedure?

### Importing APM dashboards from the Grafana community

> 

Reviewer: ```suggestion
```

> [Grafana community dashboards](promscale-grafana-dashboards) contains all the

Reviewer: ```suggestion
[Grafana community dashboards](promscale-grafana-dashboards) contain all the
```

dashboards published by Promscale. 

<procedure>

> 

Reviewer: Needs a procedure title

Reviewer 2: done

> 1.  Click the `Details` button to open a dashboard from [Grafana community

Reviewer: Needs a location ... "In the Grafana dashboard" perhaps?

    dashboard](promscale-grafana-dashboards). 
1.  Click `Copy ID to Clipboard` to copy the UID of the dashboard.
1.  In the Grafana UI, select `Import` from the `+` Create icon on the side
    menu.
1.  Paste the dashboard UID in the `Import via grafana.com` textbox, and click
    `Load`. The `Importing dashboard from Grafana.com` page appears.
1.  In  the `Folder` drop-down menu, choose the folder to which you want to add
    the dashboard.

> 1.  Select the data sources from which you want the dashboard to query the data.

Reviewer: ```suggestion
1.  Select the data sources from which you want the dashboard to query the data:
```

    * For APM dashboards select `TimescaleDB or PostgreSQL data source` as
      `Promscale-SQL`.
    * For APM dashboards select `Promscale Jaeger Tracing data source` as
      `Promscale-Tracing`.
1.  Click `Import`.

</procedure>

### Importing dashboards as JSON files.

[Promscale dashboards](promscale-github-dashboards) repository contains all the
dashboards published by Promscale. 

<procedure>

> 

Reviewer: Needs a procedure title

1.  Download all the `.json` files from the [Promscale
    dashboards](promscle-github-dashboards) repository.
1.  In the Grafana UI, select `Import` from the `+` Create icon on the side
    menu.
1.  Select the `Upload JSON file` button, and select the downloaded `JSON` dashboard
    file. The `Importing dashboard from Grafana.com` page appears.
1.  In  the `Folder` drop-down menu, choose the folder to which you want to add
    the dashboard.

> 1.  Select the data sources from which you want the dashboard to query the data.

Reviewer: ```suggestion
1.  Select the data sources from which you want the dashboard to query the data:
```

    * For APM dashboards select `TimescaleDB or PostgreSQL data source` as
      `Promscale-SQL`.
    * For APM dashboards select `Promscale Jaeger Tracing data source` as
      `Promscale-Tracing`.
1.  Click `Import`.

</procedure>

### Mounting dashboards as snapshots

> 

Reviewer: ```suggestion
```

You can mount the dashboards in Grafana on startup using the configuration
options in Grafana. For more information, see [Provisioning Grafana](grafana-dashboard-mounting).

> In that directory that is mounted to Grafana configuration, prefix `APM` to all

Reviewer: ```suggestion
In the directory that is mounted to Grafana configuration, prefix `APM` to all
```

> the dashboards with APM as prefix from [Promscale Github repository](promscale-github-dashboards). 

Reviewer: ```suggestion
the dashboards from [Promscale Github repository](promscale-github-dashboards). 
```
